### PR TITLE
Add FindGflags for supporting Gflags

### PR DESCRIPTION
**Pull request by davetcoleman**
Based on discussion here https://github.com/ros/cmake_modules/issues/36

I'm not a CMake expert (I'm ok) so I need help making this compliant with the contributions guidelines. I did my best to provide the comment documentation as requested.

This does work with several of my packages though and I'm currently using it.

### CODE REVIEW

File: cmake/Modules/FindGflags.cmake [hunk 1]
```diff
@@ -0,0 +1,66 @@
+###############################################################################
+#
+# CMake script for finding the GFlags library.
+#
+# https://gflags.github.io/gflags/
+#
+# Redistribution and use is allowed according to the terms of the 3-clause BSD
+# license.
+#
+# Input variables:
```
Comment (wjwwood) [inline cmake/Modules/FindGflags.cmake:10]: If there aren't any input variables, then drop this section?

File: cmake/Modules/FindGflags.cmake [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+# - Gflags_INCLUDE_DIRS: this is exported from somewhere else magically
+#
+# Example usage:
+#
+#   #Gflags
+#   find_package(Gflags REQUIRED)
+#
+#   include_directories(
+#     ${Gflags_INCLUDE_DIRS}
+#   )
```
Comment (wjwwood) [inline cmake/Modules/FindGflags.cmake:26]: Should the example not include usage of the `Gflags_LIBRARIES`?

File: cmake/Modules/FindGflags.cmake [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+# Redistribution and use is allowed according to the terms of the 3-clause BSD
+# license.
+#
+# Input variables:
+#
+#
+# Cache variables (not intended to be used in CMakeLists.txt files)
+#
+# - _gflags_ORIG_CMAKE_FIND_LIBRARY_SUFFIXES
+# - CMAKE_FIND_LIBRARY_SUFFIXES: ?
```
Comment (wjwwood) [inline cmake/Modules/FindGflags.cmake]: What is this section supposed to be saying?

File: cmake/Modules/FindGflags.cmake [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+#
+# Output variables:
+#
+# - Gflags_FOUND: Boolean that indicates if the package was found
+# - Gflags_LIBRARIES: shared objects exported from gflags
+# - Gflags_INCLUDE_DIRS: this is exported from somewhere else magically
```
Comment (wjwwood) [inline cmake/Modules/FindGflags.cmake]: magically?

File: cmake/Modules/FindGflags.cmake (hunk 1 above)
Comment (wjwwood) [inline cmake/Modules/FindGflags.cmake:10]: What about `Gflags_STATIC` which is used below? I'd say you need to document it here or drop that part of the code below.

File: cmake/Modules/FindGflags.cmake [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  endif(NOT Gflags_FIND_QUIETLY)
+else(Gflags_FOUND)
+  if(Gflags_FIND_REQUIRED)
+    message(FATAL_ERROR "Could not find gflags library.")
+  endif(Gflags_FIND_REQUIRED)
+endif(Gflags_FOUND)
+
+set(CMAKE_FIND_LIBRARY_SUFFIXES ${_gflags_ORIG_CMAKE_FIND_LIBRARY_SUFFIXES})
+
+set(Gflags_LIBRARIES gflags)
```
Comment (wjwwood) [inline cmake/Modules/FindGflags.cmake]: The CMake code above goes to the trouble of finding the actual `gflags` library with a call to `find_library()` and then you just set `Gflags_LIBRARIES` to `gflags` blindly. That doesn't seem right.

File: cmake/Modules/FindGflags.cmake (hunk 4 above)
Comment (wjwwood) [inline cmake/Modules/FindGflags.cmake]: Also though documented here, this variable is never set below, which I guess is why you might consider it magical, but I see no way that it gets set by this file. So I'd appreciate it if you could explain how `Gflags_INCLUDE_DIRS` gets set.

**Comment by davetcoleman** [comment]
Re: all your questions. I don't understand how any of it works, honestly. I didn't write this module, I just found it online. I'll try to revise it, though.

**Comment by davetcoleman** [comment]
I've addressed all your comments, simplified the module, and squashed the commit. Ready for re-review.

**Comment by wjwwood** [comment]
Awesome, looks good to me now. Thanks for iterating on it with me.

**Comment by wjwwood** [comment]
I'll put this on my list to release, but I need to coordinate with the others since it is such a low level package it will cause a lot of work on the farm when I release it.

**Comment by davetcoleman** [comment]
@wjwwood 

> It will not. Also I wasn't planning on releasing that change into Indigo. It's currently only on the 0.4-devel branch which is Jade only right now.

Can I also release this in Indigo?

**Comment by wjwwood** [comment]
I try not to add things to indigo. Imagine, for example, that someone has a `FindGflags.cmake` file in their package which behaves differently from the one I just merged and that they also use `cmake_modules` for something else like `eigen`. If I release this into Indigo their package could break because it might use this `FindGflags.cmake` rather than theirs. Not saying that will happen, but that is an example of something that might happen if I release change like this into Indigo. The same risk exists for Jade, but with Indigo as an LTS, I'm more comfortable taking risks with Jade.

I'll release this into Jade, and if there are no regressions I'll consider back porting it to Indigo (still not 100% safe since Indigo has way more packages than Jade).

**Pull request merged**